lldb/source/Plugins/SymbolFile/DWARF/DWARFUnit.cpp

```diff
@@ -718,13 +720,11 @@ DWARFCompileUnit *DWARFUnit::GetSkeletonUnit() {
   return llvm::dyn_cast_or_null<DWARFCompileUnit>(m_skeleton_unit);
 }
 
-void DWARFUnit::SetSkeletonUnit(DWARFUnit *skeleton_unit) {
-  // If someone is re-setting the skeleton compile unit backlink, make sure
-  // it is setting it to a valid value when it wasn't valid, or if the
-  // value in m_skeleton_unit was valid, it should be the same value.
-  assert(skeleton_unit);
-  assert(m_skeleton_unit == nullptr || m_skeleton_unit == skeleton_unit);
-  m_skeleton_unit = skeleton_unit;
+bool DWARFUnit::LinkToSkeletonUnit(DWARFUnit &skeleton_unit) {
```

Comment:
No real reason to pass in a reference here since we want the pointer. The call of this function is calling with `dwo_cu->LinkToSkeletonUnit(*this)`. Can we change `DWARFUnit &skeleton_unit` back to `DWARFUnit *skeleton_unit`?

Reply:
A reference implies (and documents) non-nullness of the argument. This removes the need for the `assert(skeleton_unit);` inside the function and makes it clear to the caller that it must pass a valid object. I don't strongly about it, but I think that's worth an extra `*` at the call site.

Reply:
Makes sense.